bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/EspIdfJsonParser.java

```diff
@@ -0,0 +1,115 @@
+/*******************************************************************************
+ * Copyright 2023 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.util;
+
+import java.io.File;
+import java.io.FileReader;
+import java.net.URL;
+import java.text.MessageFormat;
+
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.osgi.service.datalocation.Location;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+
+import com.espressif.idf.core.logging.Logger;
+
+/**
+ * esp_idf.json parser utility class to parse the 
+ * stored configs for startup in esp_idf.json
+ * @author Ali Azam Rana
+ *
+ */
+public class EspIdfJsonParser
+{
+	/**
+	 * esp-idf.json is file created by the installer
+	 */
+	private static final String ESP_IDF_JSON_FILE = "esp_idf.json"; //$NON-NLS-1$
+	
+	// Variables defined in the esp-idf.json file
+	private static final String GIT_PATH = "gitPath"; //$NON-NLS-1$
+	private static final String IDF_VERSIONS_ID = "idfSelectedId"; //$NON-NLS-1$
+	private static final String IDF_INSTALLED_LIST_KEY = "idfInstalled"; //$NON-NLS-1$
+	private static final String PYTHON_PATH = "python"; //$NON-NLS-1$
+	private static final String IDF_PATH = "path"; //$NON-NLS-1$
+	private static final String DEFAULT_WORSPACE_PATH = "defaultWorkspace"; //$NON-NLS-1$
+
+	private String gitExecutablePath;
+	private String idfVersionId;
+	private String idfPath;
+	private String pythonExecutablePath;
+	private String defaultWorkspaceLocation;
+	private boolean isIdfJsonPresent;
+
+	public void parseJsonAndLoadValues()
+	{
+		// Get the location of the eclipse root directory
+		Location installLocation = Platform.getInstallLocation();
+		URL url = installLocation.getURL();
+		Logger.log("Eclipse Install location::" + url); //$NON-NLS-1$
+		File idf_json_file = new File(url.getPath() + File.separator + ESP_IDF_JSON_FILE);
+		if (!idf_json_file.exists())
+		{
+			Logger.log(MessageFormat.format("esp-idf.json file doesn't exist at this location: '{0}'", url.getPath())); //$NON-NLS-1$
+			isIdfJsonPresent = false;
+			return;
+		}
+		
+		isIdfJsonPresent = true;
+		
+		
+		JSONParser parser = new JSONParser();
+		try
+		{
+			JSONObject jsonObj = (JSONObject) parser.parse(new FileReader(idf_json_file));
```

Comment:
we need to close the FileReader with try-with-resources and provide an explicit character set to the FileReader constructor (e.g. StandardCharsets.UTF_8) to avoid dependency on OS code.

Reply:
@sigmaaa resolved in the latest commit